File: 94. Binary Tree Inorder Traversal.js

var inorderTraversal = function(root) {

    const result = [];

    function inorder(node) {

        // Base case
        if (!node) {
            return;
        }

        // Left
        inorder(node.left);

        // Root
        result.push(node.val);

        // Right
        inorder(node.right);
    }

    inorder(root);

    return result;
};

File: 94. Binary Tree Inorder Traversal.cpp

class Solution {
public:
    vector<int> result;

    void inorder(TreeNode* root) {

        // Base case
        if (root == nullptr) {
            return;
        }

        // Traverse left subtree
        inorder(root->left);

        // Visit current node
        result.push_back(root->val);

        // Traverse right subtree
        inorder(root->right);
    }

    vector<int> inorderTraversal(TreeNode* root) {

        inorder(root);

        return result;
    }
};
